astropy/wcs/wcsapi/tests/test_fitswcs.py

```diff
@@ -1348,3 +1350,30 @@ def check_wcs(header):
     # Check fall back to scale='utc'
     del hdr["TIMESYS"]
     check_wcs(hdr)
+
+
+def test_fits_tab_time_and_units():
+    """
+    This test is a regression test for https://github.com/astropy/astropy/issues/12095
+
+    It checks the following:
+      - If a spatial WCS isn't converted to units of deg by wcslib it still works.
+      - If TIMESYS is upper case we parse it correctly
+      - If a TIME CTYPE key has an algorithm code (in this case -TAB) it still works.
+
+    The file used here was generated by gWCS and then edited to add the TIMESYS key.
+    """
+    with fits.open(Path(__file__).parent / "example_4d_tab.fits") as hdul:
```

Comment:
Wait a min... why is this not using `get_pkg_data_filename`?

Reply:
because I can't make it work :p